### PR TITLE
feat: [DEV-282] Client Library support for User Management - Node.js

### PR DESCRIPTION
**Pull request by Max-Stytch**

### CODE REVIEW

File: test/sessions.test.ts [hunk 1]
```diff
@@ -1,38 +1,6 @@
...
-      data: config.data && JSON.parse(config.data),
-    };
-    const response = handler(request);
-    return Promise.resolve({
-      ...response,
-      config,
-    });
-  };
-}
+import { mockRequest } from "./helpers";
```
Comment (jeremy-stytch) [inline test/sessions.test.ts:3]: Nice, thanks for extracting this!

**Review by jeremy-stytch — APPROVED** [review]
LGTM. I'll unblock the integration tests

File: lib/users.ts [hunk 2]
```diff
@@ -36,13 +36,120 @@ export interface CreateResponse extends BaseResponse {
...
+      filter: "email_verified";
+      email_verified: boolean;
+    }
+  | {
+      filter: "email_address_fuzzy";
+      email_address_fuzzy: string;
+    }
+  | {
+      filter: "webauthn_registration_verified";
+      webauthn_registration_verified: boolean;
```
Comment (chris-stytch) [inline lib/users.ts:127]: Missing `webauthn_registrations` filter.

File: README.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ This client library supports all of Stytch's live products:
   - [x] [Email passcodes](https://stytch.com/docs/api/send-otp-by-email)
   - [x] [Session Management (Beta)](https://stytch.com/docs/api/sessions-overview)
   - [x] [WebAuthn (Beta)](https://stytch.com/docs/api/webauthn-overview)
+  - [x] [User Management (Beta)](https://stytch.com/docs/api/users)
```
Comment (chris-stytch) [inline README.md:28]: I think I'm capitulating here, people seem to prefer sentence case for product names as well, e.g. User management.

```suggestion
  - [x] [User management (Beta)](https://stytch.com/docs/api/users)
```

File: lib/users.ts [hunk 4]
```diff
@@ -36,13 +36,126 @@ export interface CreateResponse extends BaseResponse {
...
-export interface GetResponse extends BaseResponse {
+interface User {
   user_id: UserID;
   name: Name;
   emails: Email[];
   phone_numbers: PhoneNumber[];
   webauthn_registrations: WebAuthnRegistration[];
+  providers: OAuthProvider[];
   status: string;
+  created_at: Date;
```
Comment (chris-stytch) [inline lib/users.ts:47]:
```suggestion
  user_id: UserID;
  created_at: Date;
  status: string;
  name: Name;
  emails: Email[];
  phone_numbers: PhoneNumber[];
  providers: OAuthProvider[];
  webauthn_registrations: WebAuthnRegistration[];
```

I think this order makes more "sense" if we aren't going to go with alphabetical, it is roughly status>information>auth methods.

File: test/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      params: config.params,
+      data: config.data && JSON.parse(config.data),
+    };
+    const response = handler(request);
+    return Promise.resolve({
+      ...response,
+      config,
+    });
+  };
+}
```
Comment (chris-stytch) [inline test/helpers.ts:32]: @jeremy-stytch loves having new lines for all files :)

File: dist/users.js [hunk 6]
```diff
@@ -81,4 +151,11 @@ class Users {
...
-exports.Users = Users;
\ No newline at end of file
+exports.Users = Users;
+
+function parseUser(user) {
+  console.log(user);
+  return { ...user,
+    created_at: new Date(user.created_at)
+  };
+}
```
Comment (chris-stytch) [inline dist/users.js:161]: New line

File: types/lib/users.d.ts [hunk 7]
```diff
@@ -22,13 +23,90 @@ export interface CreateResponse extends BaseResponse {
...
-export interface GetResponse extends BaseResponse {
+interface User {
     user_id: UserID;
     name: Name;
     emails: Email[];
     phone_numbers: PhoneNumber[];
     webauthn_registrations: WebAuthnRegistration[];
+    providers: OAuthProvider[];
     status: string;
+    created_at: Date;
```
Comment (chris-stytch) [inline types/lib/users.d.ts:34]: Similar re-ordering to above if we would like to do that.

File: types/lib/users.d.ts [hunk 8]
```diff
@@ -67,17 +145,29 @@ export interface DeletePhoneNumberResponse extends BaseResponse {
...
+}
 export declare class Users {
     base_path: string;
     private client;
     constructor(client: AxiosInstance);
     private endpoint;
     create(data: CreateRequest): Promise<CreateResponse>;
     get(userID: UserID): Promise<GetResponse>;
+    search(data: SearchRequest): Promise<SearchResponse>;
+    searchAll(data: SearchRequest): UserSearchIterator;
```
Comment (chris-stytch) [inline types/lib/users.d.ts:165]: For reading clarity, should we name this `searchAll`, or `getAll`? The intended purpose is to "get all users", correct? I suppose it isn't written explicitly to do get everything, just make iterating easier.

**Review by chris-stytch — APPROVED** [review]

**Comment by chris-stytch** [comment]
None of my comments are blockers. Tested it locally with no params since we're awaiting some API changes.

File: types/lib/users.d.ts (hunk 8 above)
Comment (Max-Stytch) [reply types/lib/users.d.ts:165]: the intended purpose is to handle paging through all results of a search- so.... sometimes?
 I feel like `getAll` would be better as its own helper method, eg `() => searchAll({ /* empty set of filters */ })`

File: types/lib/users.d.ts (hunk 7 above)
Comment (Max-Stytch) [reply types/lib/users.d.ts]: done (btw, `types` and `dist` are both auto-generated from the build)

File: dist/users.js (hunk 6 above)
Comment (Max-Stytch) [reply dist/users.js:161]: this file is auto-generated, so I won't think too hard about newlines here

File: lib/users.ts (hunk 4 above)
Comment (Max-Stytch) [reply lib/users.ts]: done - https://github.com/stytchauth/stytch-node/pull/97/commits/4f4f647e27f216b63d2dc6e94aac1dd10d2f1224

File: types/lib/users.d.ts (hunk 8 above)
Comment (chris-stytch) [reply types/lib/users.d.ts:165]: That makes sense to me.

**Review by grace-stytch — APPROVED** [review]

**Pull request merged**